pinot-common/src/main/java/org/apache/pinot/common/data/PinotObject.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.data;
+
+import java.util.List;
+
+/**
+ * Common interface for complex Object types such as HyperLogLog, Map, JSON etc
+ */
+public interface PinotObject {
+
+  void init(byte[] bytes);
+
+  byte[] toBytes();
+
+  /**
+   * @return
+   */
+  List<String> getPropertyNames();
```

Comment:
I guess property names and values are specific to the object type? Some objects may have properties specific to the instance (e.g. for json objects it will be "keys", I suppose) whereas others may have properties specific to the object definition?